.github/workflows/build.yml

```diff
@@ -0,0 +1,53 @@
+## Github action to run:
+# 1. Install dependencies
+# 2. Build the project
+# 3. Run tests (if any)
+# 4. Check if the version was bumped, if there are changes in the dist folder.
+
+name: Build, Test & Verify dist
+on:
+  push:
+    branches:
+      - '*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 20
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build project
+        run: npm run build
+
+      - name: Run tests
+        run: npm run test
+
+      - name: Check version bump # If there are changes in 'dist', was the package version bumped?
+        run: |
+          git fetch origin main
+
+          git diff --exit-code --name-only dist/
+
+          if [ $? -ne 0 ]; then
+            echo "New files generated in /dist folder. Please commit the changes."
+            exit 1
+          fi
+
+          package_version=$(jq -r .version package.json)
```

Comment:
should this be in the `if [ $? -ne 0 ]; then` block?

Reply:
my thinking was: 
 - if there are new files generated (e.g.: developer doesn't commit the dist/ changes) `exit 1`
 - otherwise: we can check the version bump
   - if there were changes in `dist/` compared to the merge base AND yet, the version is the same `exit 1`

If none of these `exit 1`s are hit, we're in the clear: either no changes in `dist/` or there were changes, but the version was also changed (no precise comparison on _how_ it was changed - the reviewer will check that)